libs/platform/ebpf_object.c

```diff
@@ -309,16 +342,31 @@ ebpf_object_reference_next_object(
     ebpf_object_type_t type,
     _Outptr_result_maybenull_ ebpf_core_object_t** next_object)
 {
+    ebpf_result_t return_value = EBPF_SUCCESS;
     ebpf_lock_state_t state;
     *next_object = NULL;
 
     state = ebpf_lock_lock(&_ebpf_object_tracking_list_lock);
 
     ebpf_id_t start_id = (previous_object) ? previous_object->id : 0;
-    ebpf_core_object_t* object = _get_next_object_by_id(start_id, type);
-    if (object != NULL) {
-        *next_object = object;
-        ebpf_object_acquire_reference(object);
+
+    while (*next_object == NULL) {
+        uint32_t index;
+
+        return_value = _get_next_object_id_by_type(start_id, type, &start_id);
+        if (return_value != EBPF_SUCCESS) {
+            break;
+        }
+        return_value = _get_index_from_id(start_id, &index);
+        ebpf_assert(return_value == EBPF_SUCCESS);
+        if (_ebpf_id_table[index].object == NULL) {
+            continue;
+        }
+        if (!_ebpf_object_try_acquire_reference(&_ebpf_id_table[index].object->base)) {
```

Comment:
If we fail to acquire reference, should we increment the id so that the same object is not returned again?

Reply:
The code updates start_id via _get_next_object_id_by_type and the new start_id is used to compute the new index value.